operator/v1alpha1/types_olm.go

```diff
@@ -0,0 +1,56 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// OLM provides information to configure an operator to manage the OLM controllers
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type OLM struct {
```

Comment:
Is this expected to be a singleton resource? Only ever one in the cluster and called `cluster`? Wondering if, rather than using an admission plugin for that, we could use CEL to prevent the name being anything other than `cluster` 🤔

Reply:
Ended up implementing this as a yaml patch with a `pattern` on `metadata.name`

Reply:
Yeah I would have done it as a CEL validation on the root OLM object, did you try that at all? The reason I prefer the CEL over the pattern you've added is because you can add a custom message, which is generally more helpful, but if it didn't work or came out in some odd way, I guess we can skip it